cpp/benchmarks/io/json/nested_json.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/fixture/rmm_pool_raii.hpp>
+#include <nvbench/nvbench.cuh>
+
+#include <io/json/nested_json.hpp>
+#include <io/utilities/hostdevice_vector.hpp>  //TODO find better replacement
+
+#include <tests/io/fst/common.hpp>
+
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/strings/repeat_strings.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/cuda_stream.hpp>
+#include <rmm/cuda_stream_view.hpp>
+
+#include <cstdlib>
+
+namespace cudf {
+namespace {
+auto make_test_json_data(size_type string_size, rmm::cuda_stream_view stream)
+{
+  // Test input
+  std::string input = R"(
+                      {"a":1,"b":2,"c":[3], "d": {}},
+                      {"a":1,"b":4.0,"c":[], "d": {"year":1882,"author": "Bharathi"}},
+                      {"a":1,"b":6.0,"c":[5, 7], "d": null},
+                      {"a":1,"b":null,"c":null},
+                      {
+                        "a" : 1
+                      },
+                      {"a":1,"b":Infinity,"c":[null], "d": {"year":-600,"author": "Kaniyan"}},
+                      {"a": 1, "b": 8.0, "d": { "author": "Jean-Jacques Rousseau"}},)";
+
+  const size_type repeat_times = string_size / input.size();
+
+  auto d_input_scalar   = cudf::make_string_scalar(input, stream);
+  auto& d_string_scalar = static_cast<cudf::string_scalar&>(*d_input_scalar);
+  auto d_scalar         = cudf::strings::repeat_string(d_string_scalar, repeat_times);
+  auto& d_input         = static_cast<cudf::scalar_type_t<std::string>&>(*d_scalar);
+  auto generated_json   = std::string(d_input);
```

Comment:
Wait, are you constructing a host string from device string?

Reply:
yes. only due to limitation of `parse_nested_json` at the moment. This will change once the API is complete.

Reply:
I meant you're reading device memory from the host side, which is typically illegal. Should this be invalid and cause crash?

Reply:
There is the operator `std::string` operator in `string_scalar`
https://github.com/rapidsai/cudf/blob/f94146b59fd3163be604344d40c2d3cee011d151/cpp/include/cudf/scalar/scalar.hpp#L519